src/main/java/com/netgrif/application/engine/elastic/service/ElasticTaskQueueManager.java

```diff
@@ -0,0 +1,166 @@
+            scheduleNextTask(task.getTask().getTaskId());
+        }
+    }
+
+
+    private void scheduleNextTask(String taskId) {
+        BlockingQueue<Runnable> queue = taskQueues.get(taskId);
+        if (queue != null) {
+            Runnable nextTask = queue.poll();
+            if (nextTask != null) {
+                log.debug("Submitting next task for ID: {} to executor", taskId);
+                elasticTaskExecutor.submit(nextTask);
+            } else {
+                activeTasks.remove(taskId);
+                taskQueues.remove(taskId);
+            }
+        }
+    }
+
+
+    @PreDestroy
+    public void destroy() throws InterruptedException {
+        log.info("Shutting down ElasticTaskQueueManager");
+        elasticTaskExecutor.shutdown();
+    }
+
+
+    private ElasticTask indexTaskWorker(ElasticTask task) {
+        log.debug("Indexing task [{}] in thread [{}]", task.getTaskId(), Thread.currentThread().getName());
+        ElasticTask elasticTask = null;
+        try {
+            elasticTask = repository.findByStringId(task.getStringId());
+            if (elasticTask == null) {
+                elasticTask = repository.save(task);
+            } else {
+                elasticTask.update(task);
+                elasticTask = repository.save(elasticTask);
+            }
+            log.debug("[{}]: Task \"{}\" [{}] indexed", task.getCaseId(), task.getTitle(), task.getStringId());
+        } catch (InvalidDataAccessApiUsageException e) {
+            log.debug("[{}]: Task \"{}\" has duplicates, will be reindexed", task.getCaseId(), task.getTitle());
+            repository.deleteAllByStringId(task.getStringId());
+            repository.save(task);
+            log.debug("[{}]: Task \"{}\" indexed", task.getCaseId(), task.getTitle());
+        } catch (RuntimeException e) {
+            log.error("Elastic executor was killed before finish: {}", e.getMessage());
+        }
+        return elasticTask;
+    }
+
+    private ElasticTask removeTaskWorker(ElasticTask task) {
+        log.debug("Remove task [{}] in thread [{}]", task.getTaskId(), Thread.currentThread().getName());
+        try {
+            log.debug("[{}]: Task \"{}\" [{}] removed", task.getCaseId(), task.getTitle(), task.getStringId());
+            return repository.deleteAllByTaskId(task.getTaskId());
+        } catch (RuntimeException e) {
+            log.error("Elastic executor was killed before finish: {}", e.getMessage());
+        }
+        return task;
+    }
+
+    public void removeTasksByProcess(String processId) {
+        List<ElasticTask> tasks = repository.findAllByProcessId(processId);
+        tasks.forEach(task -> {
+            try {
+                ElasticTaskJob job = new ElasticTaskJob(ElasticJob.REMOVE, task);
+                Future<ElasticTask> totok = scheduleOperation(job);
+                totok.get(30, TimeUnit.SECONDS);
+            } catch (ExecutionException e) {
```

Comment:
body of each `catch` clause is the same, java should allows combining the bodies for multiple exceptions since version 7 `(ExecutionException|interruptedException|TimeoutException e)`